### PR TITLE
Feat r custompackages

### PR DESCRIPTION
**Pull request by j-stat**
Hi, Isabel: 

Just creating a pull request so we can work together and get feedback! Specifically, we added a class (i.e., deseq_workshop_1) along with updated .yml files and Dockerfile, and created an rscipt file in scripts folder. Thank you so much! 

Kind regards, 

Jordan

### CODE REVIEW

File: scripts/r_rnaseq_packages.R
Comment (mirestrepo) [inline scripts/r_rnaseq_packages.R:7]: 1. Base R packages are installed for every R-hub... maybe we should move ggplot2 and tydr to the `requirements.txt` file so conda installs that for every R class. In fact in your case, if anything else can be installed from `conda-forge` you may want to move it there .... dependency management may be better.

2. For now  the best is to move this file to the hub specific folder i.e, `equirements/classes/deseq_workshop_1`. The contents of that folder are already available to the the docker host under `/home/$NB_USER/tmp/` - see https://github.com/brown-ccv/jupyterhub-docker-images/blob/8771e9819386e8e007102f60e5f4e0f72937ee74/Dockerfile#L108
3. Rename file to something more generic i.e `packages.R`. 
4.  Since we share 1 Dockerfile for all projects. We want to make sure, when someone generates a new class, the build wil work. So we also need to modify the `dev/templates` folder to include a "template" of this script - the template could just be your contents commented out as an example. But the post important part is that when in the docker file you call `RUN Rscript packages.R` the line works if this file is empty

File: requirements/classes/deseq_workshop_1/requirements.txt
Comment (mirestrepo) [inline requirements/classes/deseq_workshop_1/requirements.txt:49]: See long comment belo. Move other r basic packages here. i.edggplot2

File: Dockerfile
Comment (mirestrepo) [inline Dockerfile:172]: See long explanation below. Can remove the COPY line after you move the file and next line will be more like
`RUN Rscript /home/$NB_USER/tmp/packages.R`

**Review by mirestrepo — CHANGES_REQUESTED** [review]

**Comment by mirestrepo** [comment]
Also, from the build log has this warning in case it's not what you expect 
```
#34 1.313 Warning: dependencies ‘BiocStyle’, ‘BiocVersion’ are not available
11752
```
https://github.com/brown-ccv/jupyterhub-docker-images/runs/1749808463?check_suite_focus=true#step:7:11751
Build and Push Stage L11751

File: requirements/classes/deseq_workshop_1/requirements.txt
Comment (mirestrepo) [reply requirements/classes/deseq_workshop_1/requirements.txt:49]: Actually, ggplot2 is already installed by conda, maybe you have it here because you want additional dependencies? `ggplo2-mmovie` comes up as an additional one that gets installed

File: scripts/r_rnaseq_packages.R
Comment (ashokrags) [reply scripts/r_rnaseq_packages.R:7]: I think that using conda for R packages makes it worse. We want to have as much of the R packages installed from either CRAN or Bioconductor etc

File: .github/workflows/deseq_workshop_1-spring-tag.yml
Comment (mirestrepo) [inline .github/workflows/deseq_workshop_1-spring-tag.yml:5]: In the spirit of the conversation on slack, change this to `deseq` and also the file name

Comment (mirestrepo) [reply .github/workflows/deseq_workshop_1-spring-tag.yml:5]: Just be careful to change all instance in the files

File: scripts/r_rnaseq_packages.R
Comment (mirestrepo) [reply scripts/r_rnaseq_packages.R:7]: we've come a long way ... I though I was the only one that didn't like conda 😂 

In that case, you can decide and installed them as preferred.

But... Do you think that we should be generally not installing R packages from Conda? If so we could move all R packages out of the `requirements.txt` file (maybe in future PR). I don't install Julia packages via `conda` either

File: Dockerfile
Comment (mirestrepo) [inline Dockerfile:171]: @j-stat Shouldn't it just be `RUN Rscript ~packages.R~ /home/$NB_USER/tmp/packages.R`

File: dev/templates/packages.R
Comment (mirestrepo) [inline dev/templates/packages.R:1]: Can you make this comment more general, for instance
`# Use this script to install packages via CRAN, for example:`

**Review by mirestrepo — CHANGES_REQUESTED** [review]
Looking good. Just three more changes:
1. the `RUN Rscript` line needs to be fixed
2. Make the comment in the script template general
3. Add `packages.R` to classes that use `r_lang` or `r_julia` `TARGET

**Comment by mirestrepo** [comment]
The list of past classes that had R and will need to also have an empty `packages.R` is:
- [ ] neur1630
- [ ] data1010
- [ ] data2080
- [ ] econ2070
- [ ] mpa2065

**Review by mirestrepo — APPROVED** [review]
Looks great and tests are passing! I have only one tiny neat pick. But you can merge. 
There is one other thing. I had a branch that I hadn't merged to main, but just dit. You will need to rebase or update (there is a button in the GitHub UI below by the merge button) from main before these one can be merged (sorry that happened on your first PR!)

In general, once approved, I leave it up to you to merge to the `main` branch. You should have permissions, if for some reason you don't. Let me know

**Pull request merged**
